### PR TITLE
Rename to actionview html sanitizer

### PR DESCRIPTION
**Pull request by tongueroo**
## Why

This PR renames the rails-html-sanitizer to actionview-html-sanitizer.

Currently, the rails-html-sanitizer is used in actionview and defines an empty `Rails` namespace. This makes it more challenging to use with frameworks like [Jets](https://rubyonjets.com/), because once the `Rails` constant is defined, then other gems that do this:

some_gem/railtie.rb

```ruby
require "some_gem/railtie" if defined?(Rails)
```

won't be able to be used with Jets. Renaming to actionview-html-sanitizer gives other gems a chance to work with Jets, also.

Note: The ideal way to check would for the other gems to use this instead:

```ruby
require "some_gem/railtie" if defined?(Rails::Railtie)
```

It's a bummer, gems authors are not always aware of this.

## Related

* https://github.com/rails/rails/pull/49643
* https://github.com/rails/rails/pull/49714

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -24,19 +24,19 @@ Performance:
     - '**/test/**/*'
 
 # Prefer assert_not over assert !
-Rails/AssertNot:
+ActionView/AssertNot:
```
Comment (flavorjones) [inline .rubocop.yml:27]: It is not correct to change this name, these rules are defined in https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsassertnot with the `Rails` namespace.

File: lib/action_view/html/sanitizer/version.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
 # frozen_string_literal: true
 
-module Rails
+module ActionView
   module HTML
     class Sanitizer
-      VERSION = "1.6.0"
+      VERSION = "2.0.0"
```
Comment (flavorjones) [inline lib/action_view/html/sanitizer/version.rb:6]: It's inappropriate to make a version change in a feature PR.

**Review by flavorjones — COMMENTED** [review]
How do you suggest we avoid breaking existing Rails apps that are referencing constants like `Rails::Html::Sanitizer`?

I appreciate that you're trying to solve a real problem, but I do not think that renaming this project and renaming the top-level module is an appropriate solution. At the very least, you are presuming that the maintainers of this library are willing to take on a great deal of work to ensure that applications aren't broken, and I remain unconvinced that this library needs to own that responsibility.

File: .rubocop.yml (hunk 1 above)
Comment (tongueroo) [reply .rubocop.yml:27]: All good, I understand. Can workaround for Jets. Closing out. Thanks for the response though!

File: lib/action_view/html/sanitizer/version.rb (hunk 2 above)
Comment (tongueroo) [reply lib/action_view/html/sanitizer/version.rb:6]: Thank for the note.

File: .rubocop.yml (hunk 1 above)
Comment (tongueroo) [reply .rubocop.yml:27]: Oops. Ok

**Comment by tongueroo** [comment]
All good, I understand. Can workaround for Jets. Closing out. Thanks for the response though!

**Pull request closed without merging**